phoenix-core/src/main/java/org/apache/phoenix/mapreduce/PhoenixInputFormat.java

```diff
@@ -185,7 +188,10 @@ private QueryPlan getQueryPlan(final JobContext context, final Configuration con
                 scan.setAttribute(BaseScannerRegionObserver.TX_SCN, Bytes.toBytes(Long.valueOf(txnScnValue)));
             }
             // Initialize the query plan so it sets up the parallel scans
-            queryPlan.iterator(MapReduceParallelScanGrouper.getInstance());
+            if(snapshotName != null)
```

Comment:
Add an init method to the ParallelScanGrouper interface that returns an instance and have the TableMRParallelScanGrouper implementation of that return TableMRParallelScanGrouper.getInstance(). We shouldn't need switching logic here based on snapshotName.